### ISSUE TITLE
cant install on debian

### ISSUE THREAD

**Issue by brettwi** [sm0svx/svxlink#122]
svxlink: error while loading shared libraries: libasynccpp.so.1.3: cannot open shared object file: No such file or directory

this is what i get.

Would someone, anyone write a modern up to date easy to understand one line sentence (without linuxese) as to how to install on latest debian?  I can't believe all the misinformation on how to install is so completely wrong.

Thank you.

one person says to install the following packages:
apt-get install libgsm1 libgsm1-dev sigc++ libsigc++-1.2-5c2 libsigc++-1.2-dev libpopt0  libpopt-dev tcl8.4 tcl8.4-dev g++ subversion patch mutt libqt3-mt libqt3-mt-dev libqt3-headers libgcrypt11 libgcrypt11-dev libspeex1 libspeex-dev libboost-dev libasound2  ibasound2-dev

another says:
apt-get install alsa-utils alsa-oss oss-compat alsa-base libasound2-dev libasound2-plugins speex sox tcl automake autoconf libtool gcc gcc-4.7-locales gcc-4.7-multilib g++ flex bison file make cmake libsigc++-2.0-dev tcl-dev libspeex-dev libspeexdsp-dev libssl-dev libgcrypt11-dev libgsm1-dev libtclap-dev libpopt-dev libspandsp-dev checkinstall unzip screen git-all libopus-dev opus-tools swh-plugins

another says oh don't forget to:
apt-get install build-essential

and buried in the wiki it says (after having you install the program):
"don't forget to create a user and group BEFORE you install.

So would someone please step up to the plate and give simple instructions as to how to install?

I can't believe that you lot are making it impossible on purpose.

Or are you?

And yes this is an ISSUE,not a chat. It flat out does not work if you follow "instructions"

**Comment by f5vmr** [comment]
On the subject of installation, as this post is still open - I see that the latest patch is available on github.
Forgive me if I am less than intelligent, but it may be my lack of understanding of terminology, but how does one "patch" an installation? Or is it simply I download the current svxlink-master.tar.gz and install it over the current installation? Can you please explain? Regards. Chris F5VMR

**Comment by mstyne** [comment]
````
svxlink: error while loading shared libraries: libasynccpp.so.1.3: cannot open shared object file: No such file or directory
```

This can be resolved by copying the needed `.so` files into `/usr/lib` (or somewhere else appropriate):

```
$ ldd ./svxlink 
	/usr/lib/arm-linux-gnueabihf/libcofi_rpi.so (0xb6fb4000)
	libsigc-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libsigc-2.0.so.0 (0xb6f99000)
	libpopt.so.0 => /lib/arm-linux-gnueabihf/libpopt.so.0 (0xb6f86000)
	libgsm.so.1 => /usr/lib/arm-linux-gnueabihf/libgsm.so.1 (0xb6f74000)
	libtcl8.5.so.0 => /usr/lib/libtcl8.5.so.0 (0xb6e7d000)
	libgcrypt.so.11 => /lib/arm-linux-gnueabihf/libgcrypt.so.11 (0xb6dfd000)
	libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6df2000)
	libasynccpp.so.1.3 => not found
	libasyncaudio.so.1.3 => not found
	libasynccore.so.1.3 => not found
	librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb6de2000)
	libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb6dc3000)
	libspeex.so.1 => /usr/lib/arm-linux-gnueabihf/libspeex.so.1 (0xb6da5000)
	libopus.so.0 => /usr/lib/libopus.so.0 (0xb6d61000)
	libasound.so.2 => /usr/lib/arm-linux-gnueabihf/libasound.so.2 (0xb6c9d000)
	libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb6bca000)
	libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb6b59000)
	libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb6b31000)
	libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb6a00000)
	/lib/ld-linux-armhf.so.3 (0xb6fc2000)
	libgpg-error.so.0 => /lib/arm-linux-gnueabihf/libgpg-error.so.0 (0xb69f5000)
```

and running

`ldconfig -v`

Good luck!

**Comment by sm0svx** [comment]
On the subject of running the latest SvxLink master code from GitHub; It's easiest to use "git" and not the tar-archives. To get the initial version of the code:
```
git clone https://github.com/sm0svx/svxlink
```
Now you have the latest version of the code in the 'svxlink' directory. To later update to the latest version do:
```
git pull
```
Then compile and install as usual.

**Comment by pe1chl** [comment]
brettwi: probably the reason that it did not find the libraries after install is that an "ldconfig" command
has to be issued.
what exactly needs to be installed is not easy to determine for seasoned developers as they often
have things on the computer that had been installed for other reasons.
in fact the best person to write instructions for setting up is you, as you know have fresh
knowledge of the issues you encountered, what you needed to install, and what to do.  take the
opportunity to update the document now that you still remember what went wrong.  when you have
a couple of months experience with it, you will not be able to imagine why it was so difficult.

**Comment by mjrx** [comment]
So what do I need to do to get libasynccpp.so.1.4 into the right lib directory? It shows up in the build/lib but when ldconfig -v runs I see it tries to move it into user/lib/arm-linix-gnueabihf but flags "ldconfig: Cannot stat usr/lib/arm-linux-gnueabihf/libasynccpp.so.1.4: No such file or directory". I tried to manually move it but that doesn't work. Did I miss a package, somewhere, where do I get it, how do install it after everything else is built? This is turning out to be not such a trivial little project as I thought. There's got to be a simpler way to do all of this. There's dozens of installation methods out there, no two are the same. It's really frustrating me - I have a fair amount of computer expertise but I'm not a Linux guru!

**Comment by f5vmr** [comment]
Oh dear,
Sounds confused doesn’t it.
There is one one way to install it and that is to follow the directions in the svxlink.org install.doc when you click on the project – top right corner. However I did experience failure to write to certain directories if I didn’t use “sudo” before the instructions, but that will depend entirely on your directory structure permissions. When doing the cmake I followed the full instruction line sudo cmake –DCMAKE_INSTALL etc. line so that everything went to the directories I used previously on Wheezy. I now use Jessie.
I did not use ldconfig –v after completing the install, simply ldconfig. 
Don’t forget when you are copying the .so files you do need to use root permissions. Delete the old svxlink in user/lib/ when you have confirmed that the .so files now reside in /user/lib/arm-linux-gnueabihf/ folder.

Chris F5VMR
